File: app/src/main/java/com/wengtao/my_utils/SharedPreferencesUtil.java
package com.wengtao.my_utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.text.TextUtils;


/**
 * Created by wt on 2017-8-18.
 */

public class SharedPreferencesUtil {
    public static final String BASE_NAME = "SharedPreferencesUtil";

    private SharedPreferences mSharedPreferences;
    private SharedPreferences.Editor editor;

    private static SharedPreferencesUtil mSharedPreferencesUtil;

    private SharedPreferencesUtil(Context context) {
        mSharedPreferences = context.getSharedPreferences(BASE_NAME, Context.MODE_PRIVATE);
        editor = mSharedPreferences.edit();
    }

    public synchronized static SharedPreferencesUtil getInstance(Context context) {
        if (mSharedPreferencesUtil == null) {
            mSharedPreferencesUtil = new SharedPreferencesUtil(context);
        }
        return mSharedPreferencesUtil;
    }

    public void setBoolean(String key, boolean value) {
        editor.putBoolean(key, value);
        editor.commit();
    }

    public boolean getBoolean(String key) {
        return mSharedPreferences.getBoolean(key, false);
    }

    public void setInt(String key, int value) {
        editor.putInt(key, value);
        editor.commit();
    }

    public int getInt(String key) {
        return mSharedPreferences.getInt(key, 0);
    }

    public void setLong(String key, long value) {
        editor.putLong(key, value);
        editor.commit();
    }

    public long getLong(String key) {
        return mSharedPreferences.getLong(key, 0);
    }

    public void setFloat(String key, float value) {
        editor.putFloat(key, value);
        editor.commit();
    }

    public float getAFloat(String key) {
        return mSharedPreferences.getFloat(key, 0);
    }

    public void setString(String key, String value) {
        editor.putString(key, value);
        editor.commit();
    }

    public String getString(String key) {
        return mSharedPreferences.getString(key, null);
    }

    public void setObject(String key, Object value) {
        editor.putString(key, CommonUtil.objectToString(value));
        editor.commit();
    }

    public Object getObject(String key) {
        Object object = null;
        String value = mSharedPreferences.getString(key, "");
        if (!TextUtils.isEmpty(value)) {
            object = CommonUtil.stringToObject(value);
        }
        return object;
    }

}

File: app/src/main/java/com/wengtao/my_utils/SharedPreferencesUtil.kt
package com.wengtao.my_utils

import android.content.Context
import android.content.SharedPreferences
import android.text.TextUtils

class SharedPreferencesUtil private constructor(mContext: Context) {
    val BASE_TIME = "SharedPreferencesUtil"
    private var mSharedPreferences: SharedPreferences
    private var mEditor: SharedPreferences.Editor

    init {
        mSharedPreferences = mContext.getSharedPreferences(BASE_TIME, Context.MODE_PRIVATE)
        mEditor = mSharedPreferences.edit()
    }

    companion object {
        private var instance: SharedPreferencesUtil? = null

        @Synchronized
        fun getInstance(context: Context): SharedPreferencesUtil {
            if (instance == null) {
                instance = SharedPreferencesUtil(context)
            }
            return instance!!
        }
    }

    fun setBoolean(key: String, value: Boolean) {
        mEditor.putBoolean(key, value)
        mEditor.commit()
    }

    fun getBoolean(key: String): Boolean {
        return mSharedPreferences.getBoolean(key, false)
    }

    fun setInt(key: String, value: Int) {
        mEditor.putInt(key, value)
        mEditor.commit()
    }

    fun getInt(key: String): Int {
        return mSharedPreferences.getInt(key, 0)
    }

    fun setLong(key: String, value: Long) {
        mEditor.putLong(key, value)
        mEditor.commit()
    }

    fun getLong(key: String): Long {
        return mSharedPreferences.getLong(key, 0)
    }

    fun setFloat(key: String, value: Float) {
        mEditor.putFloat(key, value)
        mEditor.commit()
    }

    fun getFloat(key: String): Float {
        return mSharedPreferences.getFloat(key, 0F)
    }

    fun setString(key: String, value: String) {
        mEditor.putString(key, value)
        mEditor.commit()
    }

    fun getString(key: String): String {
        return mSharedPreferences.getString(key, null)
    }

    fun setObject(key: String, value: Any) {
        mEditor.putString(key, CommonUtil.objectToString(value))
        mEditor.commit()
    }

    fun getObject(key: String): Any? {
        var objects: Any? = null
        val value = mSharedPreferences.getString(key, "")
        if (!TextUtils.isEmpty(value)) {
            objects = CommonUtil.StringToBytes(value)
        }
        return objects
    }
}